src/tree/mod.rs

```diff
@@ -1134,6 +1248,10 @@ impl<T> Tree<T> {
     }
 }
 
+trait IteratorNew<'a, T, I> {
+    fn new(tree: &'a Tree<T>, node_id: NodeId) -> I;
+}
+
```

Comment:
Any particular reason, why you have this trait? Any reason not to make these `new` functions "normal" struct methods?